gluten-core/src/main/scala/io/glutenproject/execution/TakeOrderedAndProjectExecTransform.scala

```diff
@@ -0,0 +1,104 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.extension.TransformPreOverrides
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{Attribute, NamedExpression, SortOrder}
+import org.apache.spark.sql.catalyst.plans.physical.{Partitioning, SinglePartition}
+import org.apache.spark.sql.catalyst.util.truncatedString
+import org.apache.spark.sql.execution.exchange.ShuffleExchangeExec
+import org.apache.spark.sql.execution.{ColumnarCollapseCodegenStages, ColumnarInputAdapter,
+  SparkPlan, UnaryExecNode}
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+import java.util.concurrent.atomic.AtomicInteger
+
+case class TakeOrderedAndProjectExecTransform (
+                                                limit: Int,
+                                                sortOrder: Seq[SortOrder],
+                                                projectList: Seq[NamedExpression],
+                                                child: SparkPlan) extends
+  UnaryExecNode {
+  override def outputPartitioning: Partitioning = SinglePartition
+  override def outputOrdering: Seq[SortOrder] = sortOrder
+  override def supportsColumnar: Boolean = true
+
+  override def output: Seq[Attribute] = {
+    projectList.map(_.toAttribute)
+  }
+
+  override def simpleString(maxFields: Int): String = {
+    val orderByString = truncatedString(sortOrder, "[", ",", "]", maxFields)
+    val outputString = truncatedString(output, "[", ",", "]", maxFields)
+
+    s"TakeOrderedAndProjectExecTransform(limit=$limit, " +
+      s"orderBy=$orderByString, output=$outputString)"
+  }
+
+  override protected def withNewChildInternal(newChild: SparkPlan):
+    TakeOrderedAndProjectExecTransform =
+    copy(child = newChild)
+
+  protected override def doExecute(): RDD[InternalRow] = {
+    throw new UnsupportedOperationException(s"This operator doesn't support doExecute().")
+  }
+
+
+  protected override def doExecuteColumnar(): RDD[ColumnarBatch] = {
+    val childRDD = child.executeColumnar()
+    val childRDDPartsNum = childRDD.getNumPartitions
+    if (childRDDPartsNum == 0) {
+      // ParallelCollectionRDD is private in spark, cannot be used here.
+      // But this is not a normal usage.
+      throw new UnsupportedOperationException(s"This operator doesn't support empty partitions")
+    } else {
+      val codegenStageCounter: AtomicInteger = ColumnarCollapseCodegenStages.codegenStageCounter
+      val sortExecPlan = SortExecTransformer(sortOrder, false, new ColumnarInputAdapter(child))
```

Comment:
`child` already is a `WholeStageTransformerExec`, can remove this `WholeStageTransformerExec` and then add a new one for sort, limit ?